tests/infrastructure/node-labeller.go

```diff
@@ -203,15 +203,22 @@ var _ = DescribeInfra("Node-labeller", func() {
 		})
 
 		It("[test_id:6247] should set default obsolete cpu models filter when obsolete-cpus-models is not set in kubevirt config", func() {
+			kvConfig := v1.KubeVirtConfiguration{ObsoleteCPUModels: nil}
+			tests.UpdateKubeVirtConfigValueAndWait(kvConfig)
 			node := nodesWithKVM[0]
-
-			for key := range node.Labels {
-				if strings.Contains(key, v1.CPUModelLabel) {
-					model := strings.TrimPrefix(key, v1.CPUModelLabel)
-					Expect(nodelabellerutil.DefaultObsoleteCPUModels).ToNot(HaveKey(model),
-						"Node can't contain label with cpu model, which is in default obsolete filter")
+			Eventually(func() error {
+				node, err = virtClient.CoreV1().Nodes().Get(context.Background(), node.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
AFAIU, the test will immediately fail in case there was an error fetching the node.
Please consider returning an error, so there will be a retry.
Also, please consider giving the fetched node another name, so the details of the original node will not be lost in case of an error.

Reply:
the Get API call shouldn't fail.

Reply:
There could be a temporary network hiccup.

Reply:
In that case i don't think we should retry.

Reply:
Why not?
Failing immediately due to an API call failure could make this test flaky.

Reply:
In any other place in the functional tests, I never saw that we retry if the API call fails. I prefer to keep it consistent.

Reply:
I tend to agree with @orelmisan!
This is an `Eventually` block and we should satisfy the "eventually" behavior.
As is, if an error occurs the block is never retried.
I suggest you to switch to:
```
Eventually(func(g Gomega) {
	...
	g.Expect(err).ToNot(HaveOccurred())
	...
```
Or, as suggested, to return the error, so that it will be wrapped by the `Eventually`.
IMHO if there are other places where things are done like this, we should fix them instead of spreading it. :)
Thank you!

Reply:
@fossedihelm I disagree. 
Eventually doesn't fit all cases. For instance, in the scenario of a connectivity issue causing the get API call to fail, I would prefer to know about it as soon as possible. Such failures might suggest other issues due to instability. The 'eventually' scope should be used in cases where we expect components to eventually become consistent. In the mentioned scenario, the 'eventually' scope fits perfectly because we change the configuration and wait for the node labeller to propagate the new configuration and label the nodes accordingly. Additionally, if we would retry every function that returns an error, the code would include significant amount boilerplate.

Reply:
All over the e2e test suite, we do API calls like `Get()` or even `Create()` and expect no error occurred, without giving it a chance to try again.
I think I'm with Barak here and don't think we should necessarily keep looping if an API call failed...
As Orel mentioned, returning the error would make this test slightly more robust against flaky infra, but I don't think tests are responsible for doing that.
This is an interesting debate though. Once we reach an agreement, we should document the decision as a KubeVirt coding convention (in that doc maybe? https://github.com/kubevirt/kubevirt/pull/11456)

Reply:
Sure! Agree with you that we should reach an agreement and align all the cases to it.
Thank you